code/modules/tents/deployed_tents.dm

```diff
@@ -77,22 +78,55 @@
 /obj/structure/tent/attack_alien(mob/living/carbon/xenomorph/M)
 	if(unslashable)
 		return
-	health -= 20
+
+	M.animation_attack_on(src)
+	health -= rand(M.melee_damage_lower, M.melee_damage_upper)
+	playsound(src, 'sound/items/paper_ripped.ogg', 25, 1)
+
+	M.visible_message(SPAN_DANGER("[M] [M.slashes_verb] [src]!"), \
+	SPAN_DANGER("You [M.slash_verb] [src]!"), null, 5, CHAT_TYPE_XENO_COMBAT)
+
 	if(health <= 0)
 		visible_message(SPAN_BOLDWARNING("The [src] collapses!"))
-		qdel(src)
+		qdel(src)//need to unregister all the signal first? turn off roof plane for everyone inside?
+
+	return XENO_ATTACK_ACTION
+
+/obj/structure/tent/attackby(obj/item/item, mob/user)
+	var/obj/item/tool/shovel/shovel = item
+	if(!istype(shovel) || shovel.folded)
+		return
+	visible_message(SPAN_HIGHDANGER("[user] is trying to tear down the [src]"))
+	playsound(src, 'sound/items/paper_ripped.ogg', 25, 1)
+
+	if(user.action_busy || !do_after(user, 150, INTERRUPT_ALL|BEHAVIOR_IMMOBILE, BUSY_ICON_HOSTILE, src) || QDELETED(src))
+		return
+
+	visible_message(SPAN_HIGHDANGER("[user] tears down the [src]"))
+	playsound(src, 'sound/items/paper_ripped.ogg', 25, 1)
+	qdel(src)
+
+/obj/structure/tent/get_projectile_hit_boolean(obj/item/projectile/P)
+	. = ..()
+	return FALSE // Always fly through the tent
+
+/obj/structure/tent/Destroy()
+	for(var/turf/turf in locs)
+		for(var/mob/mobs in turf)
+			mob_exited_tent(mobs)
```

Comment:
this should be handled as another signal registered during movable_entering_tent